examples/websockets/src/main.rs

```diff
@@ -59,58 +76,169 @@ async fn main() {
+                .await
+                .is_err()
+            {
+                break;
+            }
+
+            tokio::time::sleep(std::time::Duration::from_millis(300)).await;
+            i -= 1;
+        }
+        if i == 0 {
+            println!("Sending close to {}...", who);
+            if let Err(e) = sender
+                .send(Message::Close(Some(CloseFrame {
+                    code: 1000,
+                    reason: Cow::from("Goodbye"),
+                })))
+                .await
+            {
+                println!("Could not send Close due to {}, probably it is ok?", e);
+            }
+        }
+        i
+    });
+
+    // This second task will receive messages from client and print them on server console
+    let mut recv_task = tokio::spawn(async move {
+        let mut cnt = 0;
+        while let Some(Ok(msg)) = receiver.next().await {
+            cnt += 1;
+            // print message and break if instructed to do so
+            if process_message(msg, who) {
+                break;
+            }
+        }
+        cnt
+    });
+
+    // If any one of the tasks exit, abort the other.
+    tokio::select! {
+        rv_a = (&mut send_task) => {
+            match rv_a {
+                Ok(a)=>println!("{} messages remaining to send", a),
+                Err(a)=>println!("Error sending messages {:?}", a)
+            }
+            recv_task.abort();
+        },
+        rv_b = (&mut recv_task) =>{
+            match rv_b {
+                Ok(b)=>println!("Received {} messages", b),
+                Err(b)=>println!("Error receiving messages {:?}", b)
+            }
+            send_task.abort();
+        }
+    }
+
+    // returning from the handler destroys the websocket context
+    println!("Websocket context {} destroyed", who);
+}
+
+/// helper to print contents of messages to stdout. special treatment for Close
+fn process_message(msg: Message, who: SocketAddr) -> bool {
+    match msg {
+        Message::Text(t) => {
+            println!(">>> {} sent str: {:?}", who, t);
+        }
+        Message::Binary(d) => {
+            println!(">>> {} sent {} bytes: {:?}", who, d.len(), d);
+        }
+        Message::Close(c) => {
+            if let Some(cf) = c {
+                println!(
+                    ">>> {} sent close with code {} and reason `{}`",
+                    who, cf.code, cf.reason
+                );
+            } else {
+                println!(">>> {} somehow sent close message without CloseFrame", who);
+            }
+            return true;
+        }
+
+        Message::Pong(v) => {
+            println!(">>> {} sent pong with {:?}", who, v);
+        }
+        // You should never need to manually handle these, as tungstenite websocket library
+        // will do so for you automagically by replying with Pong and copying the v according to
+        // spec. But if you need the contents of the pings you can see them here.
```

Comment:
Either we should explain what tungstenite is (the websocket library axum uses internally) or say that axum handles pings.